packages/react/src/components/Button/Button.tsx

```diff
@@ -147,21 +152,31 @@ class Button extends UIComponent<WithAsProp<ButtonProps>, ButtonState> {
         {Box.create(!hasChildren && content, {
           defaultProps: { as: 'span', styles: styles.content },
         })}
-        {!hasChildren && iconPosition === 'after' && this.renderIcon(variables, styles)}
+        {!hasChildren && iconPosition === 'after' && !loading && this.renderIcon(variables, styles)}
       </ElementType>
     )
   }
 
   renderIcon = (variables, styles) => {
-    const { icon, iconPosition, content } = this.props
-
-    return Icon.create(icon, {
-      defaultProps: {
-        styles: styles.icon,
-        xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
-        variables: variables.icon,
-      },
-    })
+    const { icon, iconPosition, content, loading } = this.props
+
+    return loading
+      ? Loader.create(
```

Comment:
Should we add slot of this? Should we make the loading shorthand for the Loader component?